### PR TITLE
Model: add to_dict and from_dict

### PR DESCRIPTION
**Pull request by davidpricedev**
The existing to_json and from_json functions already handled everything necessary for to_dict and from_dict, so this ends up being a fairly trivial change. 

I believe existing tests for to_json and from_json will exercise these changes adequately. 

Should solve #1031

### CODE REVIEW

**Review by ikonst — COMMENTED** [review]
Seems reasonable.

I'd add some docstrings. Is there an accepted AWS nomenclature for this JSON representation? From this [SO answer](https://stackoverflow.com/a/47664534), I see there's a "serialization to JSON" in AWS's own SDKs. Does the serialization we produce here matches theirs?

Basically I want folks to know there's the `{"foo": "bar"}` representation and there's the `{'foo": {"S": "bar"}}` representation, and that they're different and when you're getting which.

**Comment by davidpricedev** [comment]
@ikonst, good call, I added some docstrings, how does that look?

File: pynamodb/models.py [hunk 1]
```diff
@@ -1109,23 +1109,48 @@ def _serialize_keys(cls, hash_key, range_key=None) -> Tuple[_KeyType, _KeyType]:
     def serialize(self, null_check: bool = True) -> Dict[str, Dict[str, Any]]:
         """
         Serialize attribute values for DynamoDB
+        See the `to_dict` method to get a non-DynamoDB dict
```
Comment (ikonst) [inline pynamodb/models.py:1112]:
```suggestion
        Serialize attribute values for DynamoDB API.
        See :func:`~pynamodb.models.Model.to_dict` for a simple JSON-serializable dict.
```

File: pynamodb/models.py [hunk 2]
```diff
@@ -1109,23 +1109,48 @@ def _serialize_keys(cls, hash_key, range_key=None) -> Tuple[_KeyType, _KeyType]:
...
         """
         return self._container_serialize(null_check=null_check)
 
     def deserialize(self, attribute_values: Dict[str, Dict[str, Any]]) -> None:
         """
         Sets attributes sent back from DynamoDB on this object
+        See the `from_dict` method to fill in attributes from a non-DynamoDB dict
```
Comment (ikonst) [inline pynamodb/models.py:1119]:
```suggestion
        Sets attributes sent back from DynamoDB on this object.
        Use :func:`~pynamodb.models.Model.from_dict` to set fields from a dictionary
        previously produced by :func:`~pynamodb.models.Model.to_dict`.
```

File: pynamodb/models.py [hunk 3]
```diff
@@ -1109,23 +1109,48 @@ def _serialize_keys(cls, hash_key, range_key=None) -> Tuple[_KeyType, _KeyType]:
...
         """
         return self._container_deserialize(attribute_values=attribute_values)
 
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Converts the contents of this instance into a regular Python dict.
+        See the `serialize` method if you need the DynamoDB formated dict
```
Comment (ikonst) [inline pynamodb/models.py:1126]:
```suggestion
        Returns the contents of this instance as a JSON-serializable dict.
        See :func:`~pynamodb.models.Model.serialize` if you need to serialize
        into a DynamoDB record.
```

File: pynamodb/models.py [hunk 4]
```diff
@@ -1109,23 +1109,48 @@ def _serialize_keys(cls, hash_key, range_key=None) -> Tuple[_KeyType, _KeyType]:
...
+        """
+        return {k: attribute_value_to_json(v) for k, v in self.serialize().items()}
+
     def to_json(self) -> str:
-        return json.dumps({k: attribute_value_to_json(v) for k, v in self.serialize().items()})
+        """
+        Converts the contents of this instance into a regular Python dict.
+        See the `serialize` method if you need the DynamoDB formated dict
```
Comment (ikonst) [inline pynamodb/models.py:1133]:
```suggestion
        Returns the contents of this instance as serialized JSON (not in DynamoDB Record format).
```

File: pynamodb/models.py [hunk 5]
```diff
@@ -1109,23 +1109,48 @@ def _serialize_keys(cls, hash_key, range_key=None) -> Tuple[_KeyType, _KeyType]:
...
+        """
+        return json.dumps(self.to_dict())
 
-    def from_json(self, s: str) -> None:
-        attribute_values = {k: json_to_attribute_value(v) for k, v in json.loads(s).items()}
+    def from_dict(self, d: Dict[str, Any]) -> None:
+        """
+        Fills in properties on this instance from the provided regular python dict
+        See the `deserialize` method if the input is in DynamoDB format
```
Comment (ikonst) [inline pynamodb/models.py:1140]:
```suggestion
        Sets attributes from a dict previously produced by :func:`~pynamodb.models.Model.to_dict`.
        Use :func:`~pynamodb.models.Model.deserialize` if the dict is a DynamoDB Record
        (e.g. from a DynamoDB API response or DynamoDB Streams).
```

File: pynamodb/models.py [hunk 6]
```diff
@@ -1109,23 +1109,48 @@ def _serialize_keys(cls, hash_key, range_key=None) -> Tuple[_KeyType, _KeyType]:
...
+        """
+        attribute_values = {
+            k: json_to_attribute_value(v) for k, v in d.items()}
         self._update_attribute_types(attribute_values)
         self.deserialize(attribute_values)
 
+    def from_json(self, s: str) -> None:
+        """
+        Fills in properties on this instance from the provided regular json string
+        See the `deserialize` method if the input is in DynamoDB format
```
Comment (ikonst) [inline pynamodb/models.py:1150]:
```suggestion
        Sets attributes from a dict previously produced by :func:`~pynamodb.models.Model.to_json`.
```

**Review by ikonst — APPROVED** [review]

**Comment by ikonst** [comment]
Thanks @davidpricedev for pushing this through!

**Comment by davidpricedev** [comment]
Thanks for your help!

**Pull request merged**